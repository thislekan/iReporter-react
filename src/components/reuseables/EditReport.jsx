import React from 'react';
import moment from 'moment';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import cx from 'classnames';
import PropTypes from 'prop-types';
import style from '../../styles/editReport.css';

function EditReport(props) {
  const {
    incident,
    alterDisableComment,
    alterDisableLocation,
    disableComment,
    disableLocation,
    handleChange,
    comment,
    incidentLocation,
    location,
    disableStatus,
    alterDisableStatus,
    incidentStatus,
  } = props;
  return (
    <div className={style.editReport}>
      <div className={style.createReport}>
        <p><b>Edit an incident</b></p>
        <div className="edit__body">
          <div className="edit__body__wrapper">
            <div className={style['report-details']}>
              <div className={style['incident-intro']}>
                <FontAwesomeIcon icon='check-circle' />
                <h4>Title: </h4>
              </div>
              <p id="incident-title">{incident.title}</p>
            </div>
            <div className={style['report-details']}>
              <div className={style['incident-intro']}>
                <FontAwesomeIcon icon='check-circle' />
                <h4>Type: </h4>
              </div>
              <p id="incident-type">{incident.type}</p>
            </div>
            <div className={style['report-details']}>
              <div className={style['incident-intro']}>
                <FontAwesomeIcon icon='check-circle' />
                <h4>Created: </h4>
              </div>
              <p id="incident-date">{moment(Number(incident.createdOn)).format('MMMM Do YYYY')}</p>
            </div>
            {
              (location.pathname !== '/admin') ? null : <div className={style['report-details']}>
                <div className={style['incident-intro']}>
                  <FontAwesomeIcon icon='check-circle' />
                  <h4>Status: </h4>
                </div>
                <select
                  name="status"
                  id="incident-status"
                  onChange={handleChange}
                  onBlur={handleChange}
                  disabled={disableStatus}
                  value={incidentStatus}
                >
                  <option value="">Select status</option>
                  <option value="draft">Draft</option>
                  <option value="under-investigation">Under Investigation</option>
                  <option value="rejected">Rejected</option>
                  <option value="resolved">Resolved</option>
                </select>
              </div>
            }
            <div className={style['location-div']}>
              <div className={style['comment-div']}>
                <h4>Location: </h4>
                <textarea
                  name="location"
                  id="incident-location"
                  disabled={disableLocation}
                  value={incidentLocation}
                  onChange={handleChange}
                />
              </div>
            </div>
            <div className={cx(style['comment-div'], style['location-div'])}>
              <h4>Comment</h4>
              <textarea
                type="text"
                name="comment"
                id="incident-comment"
                disabled={disableComment}
                value={comment}
                onChange={handleChange}
              />
            </div>
            <div className={style['edit-btn-div']}>
              {
                (location.pathname === '/admin') ? null : <button
                  className={cx(style['edit-btn'], style['edit-comment'])}
                  id="edit-comment-btn"
                  onClick={alterDisableComment}
                >
                  Edit Comment
              </button>
              }
              {
                (location.pathname === '/admin') ? null : <button
                  className={cx(style['edit-btn'], style['edit-location'])}
                  id="edit-location-btn"
                  onClick={alterDisableLocation}
                >
                  Edit Location
                </button>
              }
              {
                (location.pathname !== '/admin') ? null : <button
                  className={cx(style['edit-btn'], style['edit-comment'])}
                  id="edit-location-btn"
                  onClick={alterDisableStatus}
                >
                  Edit Status
              </button>
              }
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

EditReport.propTypes = {
  incident: PropTypes.object.isRequired,
  alterDisableComment: PropTypes.func.isRequired,
  alterDisableLocation: PropTypes.func.isRequired,
  disableComment: PropTypes.bool.isRequired,
  disableLocation: PropTypes.bool.isRequired,
  handleChange: PropTypes.func.isRequired,
  comment: PropTypes.string.isRequired,
  incidentLocation: PropTypes.string.isRequired,
  location: PropTypes.object.isRequired,
  disableStatus: PropTypes.bool.isRequired,
  alterDisableStatus: PropTypes.func.isRequired,
  incidentStatus: PropTypes.string.isRequired,
};

export default EditReport;
