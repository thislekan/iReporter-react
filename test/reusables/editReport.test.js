import React from 'react';
import { shallow } from 'enzyme';
import EditReport from '../../src/components/reuseables/EditReport.jsx';
import incident from '../mock/incident';

const mock = incident;
const mockFn = jest.fn();

describe('<SingleReport />', () => {
  it('should render EditReport', () => {
    const wrapper = shallow(
      <EditReport
        incident={mock}
        location={{ pathname: '/user' }}
        incidentLocation={mock.location}
        comment={mock.comment}
        alterDisableComment={mockFn}
        alterDisableLocation={mockFn}
        alterDisableStatus={mockFn}
        incidentStatus={mock.status}
        disableComment
        disableStatus
        disableLocation
        handleChange={mockFn}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a different content when pathname is Admin', () => {
    const wrapper = shallow(
      <EditReport
        incident={mock}
        location={{ pathname: '/admin' }}
        incidentLocation={mock.location}
        comment={mock.comment}
        alterDisableComment={mockFn}
        alterDisableLocation={mockFn}
        alterDisableStatus={mockFn}
        incidentStatus={mock.status}
        disableComment
        disableStatus
        disableLocation
        handleChange={mockFn}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
