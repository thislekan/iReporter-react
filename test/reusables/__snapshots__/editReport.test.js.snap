// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleReport /> should render EditReport 1`] = `
<div>
  <div>
    <p>
      <b>
        Edit an incident
      </b>
    </p>
    <div
      className="edit__body"
    >
      <div
        className="edit__body__wrapper"
      >
        <div>
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <h4>
              Title: 
            </h4>
          </div>
          <p
            id="incident-title"
          >
            dhjfd
          </p>
        </div>
        <div>
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <h4>
              Type: 
            </h4>
          </div>
          <select
            disabled={false}
            id="incident-type"
            name="incidentType"
            onBlur={[MockFunction]}
            onChange={[MockFunction]}
            value="red-flag"
          >
            <option
              value=""
            >
              Select type
            </option>
            <option
              value="red-flag"
            >
              Red flag
            </option>
            <option
              value="intervention"
            >
              Intervention
            </option>
          </select>
        </div>
        <div>
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <h4>
              Created: 
            </h4>
          </div>
          <p
            id="incident-date"
          >
            Invalid date
          </p>
        </div>
        <div>
          <div>
            <h4>
              Location: 
            </h4>
            <textarea
              disabled={true}
              id="incident-location"
              name="location"
              onChange={[MockFunction]}
              value="sdjnkjs"
            />
          </div>
        </div>
        <div
          className=""
        >
          <h4>
            Comment
          </h4>
          <textarea
            disabled={true}
            id="incident-comment"
            name="comment"
            onChange={[MockFunction]}
            type="text"
            value="jhdfhfdjk"
          />
        </div>
        <div>
          <button
            className=""
            id="edit-comment-btn"
            onClick={[MockFunction]}
          >
            Edit Comment
          </button>
          <button
            className=""
            id="edit-location-btn"
            onClick={[MockFunction]}
          >
            Edit Location
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SingleReport /> should render a different content when pathname is Admin 1`] = `
<div>
  <div>
    <p>
      <b>
        Edit an incident
      </b>
    </p>
    <div
      className="edit__body"
    >
      <div
        className="edit__body__wrapper"
      >
        <div>
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <h4>
              Title: 
            </h4>
          </div>
          <p
            id="incident-title"
          >
            dhjfd
          </p>
        </div>
        <div>
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <h4>
              Type: 
            </h4>
          </div>
          <select
            disabled={true}
            id="incident-type"
            name="incidentType"
            onBlur={[MockFunction]}
            onChange={[MockFunction]}
            value="red-flag"
          >
            <option
              value=""
            >
              Select type
            </option>
            <option
              value="red-flag"
            >
              Red flag
            </option>
            <option
              value="intervention"
            >
              Intervention
            </option>
          </select>
        </div>
        <div>
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <h4>
              Created: 
            </h4>
          </div>
          <p
            id="incident-date"
          >
            Invalid date
          </p>
        </div>
        <div>
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="check-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <h4>
              Status: 
            </h4>
          </div>
          <select
            disabled={true}
            id="incident-status"
            name="status"
            onBlur={[MockFunction]}
            onChange={[MockFunction]}
            value="draft"
          >
            <option
              value=""
            >
              Select status
            </option>
            <option
              value="draft"
            >
              Draft
            </option>
            <option
              value="under-investigation"
            >
              Under Investigation
            </option>
            <option
              value="rejected"
            >
              Rejected
            </option>
            <option
              value="resolved"
            >
              Resolved
            </option>
          </select>
        </div>
        <div>
          <div>
            <h4>
              Location: 
            </h4>
            <textarea
              disabled={true}
              id="incident-location"
              name="location"
              onChange={[MockFunction]}
              value="sdjnkjs"
            />
          </div>
        </div>
        <div
          className=""
        >
          <h4>
            Comment
          </h4>
          <textarea
            disabled={true}
            id="incident-comment"
            name="comment"
            onChange={[MockFunction]}
            type="text"
            value="jhdfhfdjk"
          />
        </div>
        <div>
          <button
            className=""
            id="edit-location-btn"
            onClick={[MockFunction]}
          >
            Edit Status
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
